src/test/groovy/unit/org/pih/warehouse/api/ProductApiTest.groovy

```diff
@@ -0,0 +1,178 @@
+                setAccept(ContentType.JSON).build()
+    }
+
+    void "should return a list of products"() {
+        given: "an authenticated user"
+        RequestSpecification request = given().spec(requestSpecification).log().all()
+
+        when: "we get products from the API"
+        // FIXME figure out better way to set up baseURI using RestAssured API
+        Response response = request.when().get("/api/products?max=10")
+
+        // FIXME Should be replaced by Product domain class once we
+        List<ProductResponse> products = response.then().log().all().extract().body().jsonPath().
+                getList("data", ProductResponse.class)
+
+        then: "expected a 200 response with at most 10 products"
+        response.then().assertThat().statusCode(200)
+        assertThat(products, is(notNullValue()))
+        assertThat(products.size(), is(lessThanOrEqualTo(10)))
+
+        // Get a single product to pass to the next test
+        when: "we attempt to share a product with the next test"
+        ProductResponse product = response.then().body().extract().jsonPath().getObject("data[0]", ProductResponse)
+        productId = product.id
+
+        then: "product should not be null"
+        assertThat(productId, is(notNullValue()))
+
+    }
+
+    void "get demand of product"() {
+        given: "api/products/${productId}/demand called"
+        RequestSpecification httpRequest = given().spec(requestSpecification).
+                pathParams("productId", productId)
+
+        when: "we request product demand resource"
+        Response response = httpRequest.get("/api/products/{productId}/demand")
+        JsonPath jsonPathEvaluator = response.jsonPath()
+        def data = jsonPathEvaluator.get("data")
+
+        then:
+        response.then().assertThat().statusCode(200)
+
+//        then: "data should not be null"
+//        Assert.assertNotNull(data)
+//        and: "product should not be null"
+//        Assert.assertNotNull(data.product, "Response should have product object")
+//        and: "product ID should not be null"
+//        Assert.assertEquals(data.product.id, productId, "Product id should match with requested id:${productId}")
+//        and: "demand should not be null"
+//        Assert.assertNotNull(data.demand, "Demand Object should not be null")
+    }
+
+    void "get demand summary of product"() {
+        given:
+        RequestSpecification httpRequest = given().spec(requestSpecification).
+                pathParams("productId", productId)
+
+        when:
+        Response response = httpRequest.get("/api/products/{productId}/demandSummary")
+        // Get JSON Representation from Response Body
+        JsonPath jsonPathEvaluator = response.jsonPath()
+        def data = jsonPathEvaluator.getList("data")
+
+        then:
+        response.then().assertThat().statusCode(200)
+//        Assert.that(is(notNullValue(data)))
+//        and:
+//        Assert.that(is(data.size(), equalTo(0)))
```

Comment:
should it removed/uncommented?

Reply:
@chetanmaharishi can you turn these into valid rest-assured assert thingees like the first example I did.